spec/libmemcached_spec.lua

```diff
@@ -2,6 +2,18 @@ local libmemcached = require 'libmemcached'
 local json = require 'dkjson'
 local assert, describe, it = assert, describe, it
 
+local function count_used_registry()
+    local r = debug.getregistry()
+    local n = 0
+    for key, value in pairs(r) do
+        if type(key) == 'number' and type(value) ~= 'number' then
```

Comment:
Keys of lua refs are numbers, values are arbitrary types. If a ref was deleted, its value is replaced with a number, which is used to maintain linked list. I used this fact to count refs that were not deleted.